components/style/build_gecko.rs

```diff
@@ -43,42 +43,32 @@ mod bindings {
     use super::common::*;
     use super::super::PYTHON;
     use toml;
+    use toml::value::Table;
 
     const STRUCTS_FILE: &'static str = "structs.rs";
     const BINDINGS_FILE: &'static str = "bindings.rs";
 
-    fn read_config(path: &PathBuf) -> toml::Table {
+    fn read_config(path: &PathBuf) -> Table {
         println!("cargo:rerun-if-changed={}", path.to_str().unwrap());
         update_last_modified(&path);
 
         let mut contents = String::new();
         File::open(path).expect("Failed to open config file")
             .read_to_string(&mut contents).expect("Failed to read config file");
-        let mut parser = toml::Parser::new(&contents);
-        if let Some(result) = parser.parse() {
-            result
-        } else {
-            use std::fmt::Write;
-            let mut reason = String::from("Failed to parse config file:");
-            for err in parser.errors.iter() {
-                let parsed = &contents[..err.lo];
-                write!(&mut reason, "\n* line {} column {}: {}",
-                       parsed.lines().count(),
-                       parsed.lines().last().map_or(0, |l| l.len()),
-                       err).unwrap();
-            }
-            panic!(reason)
+        match contents.parse::<toml::Value>() {
```

Comment:
AFAICT you can use `toml::from_str` directly with `toml::Table` and avoid the nondescript `unwrap` call.